sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -193,10 +198,71 @@ private object CosmosRowConverter
         }
     }
 
+    private def getFullFidelityMetadata(objectNode: ObjectNode): Option[ObjectNode] = {
+      if (objectNode == null) {
```

Comment:
Don't think this should return an Option, but rather the ObjectNode, otherwise the Seq on convertStructToSparkDataType would have an Option and not the actual value, and that might nor go well with the Schema?

This is used when the conversion is Json -> Spark row, so it needs to be treated as if it was an objectnode.

In that case, I believe it should return the `toString` of the metadata object node.

```
val metadata = objectNode.get(FullFidelityChangeFeedMetadataPropertyName)
        if (metadata != null && metadata.isObject) {
          metadata.asInstanceOf[ObjectNode].toString
        } else {
          null
        }
```